### PR TITLE
Fixes schema generation error with subscriptions excluded

### PR DESCRIPTION
**Pull request by a-alle**
# Description

Fixes schema generation error when types have relationships to (subscriptions) excluded types or no (subscriptions) non-excluded fields, for both standard and abstract types. 
Adds schema tests 

## Complexity

Complexity:  Low/Medium

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2833.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
| 🟥 query.QueryWithNestedIn | 13526 | 12153 | 74 | 80 | 1610 |


🟥 - Performance worsened (dbHits)
🟩 - Performance improved (dbHits)
🟦 - New test
        

<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 39 | 60 | 1134 |
|  aggregations.NestedAggregation | 13514 | 13514 | 86 | 143 | 2174 |
|  aggregations.AggregationWithWhere | 10942 | 10942 | 70 | 86 | 2174 |
|  aggregations.NestedCountFromMovieToActors | 8694 | 8694 | 60 | 71 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 8900 | 8900 | 70 | 69 | 2174 |
|  aggregations.DeeplyNestedCount | 10054408 | 10054408 | 4264 | 4622 | 2008534 |
|  batch-create.BatchCreate | 3600 | 3600 | 98 | 103 | 600 |
|  connect.createAndConnect | 12419 | 12419 | 167 | 289 | 3003 |
|  connections.Connection | 13042 | 13042 | 95 | 125 | 2174 |
|  connections.NestedConnection | 33883 | 33883 | 110 | 131 | 4516 |
|  connections.ConnectionWithSort | 2277 | 2277 | 67 | 94 | 1040 |
|  connections.ConnectionWithSortAndCypher | 2277 | 2277 | 195 | 310 | 1040 |
|  create.SimpleMutation | 6 | 6 | 30 | 36 | 1 |
|  cypher-directive.TopLevelCypherDirective | 99525 | 99525 | 738 | 1070 | 10168 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 127992 | 127992 | 299 | 364 | 10168 |
|  delete.SimpleDelete | 18361 | 18361 | 226 | 227 | 1040 |
|  delete.NestedDeleteInUpdate | 16779 | 16779 | 153 | 173 | 2040 |
|  query.SimpleQuery | 14120 | 14120 | 102 | 225 | 2174 |
|  query.QueryWhere | 8682 | 8670 | 36 | 41 | 2168 |
|  query.SimpleQueryWithNestedWhere | 8853 | 8841 | 52 | 59 | 2168 |
|  query.Nested | 10084290 | 10084290 | 11261 | 10035 | 2008534 |
|  query.NestedWithFilter | 37172 | 37172 | 121 | 185 | 4511 |
|  query.OrFilterOnRelationships | 36780 | 35279 | 159 | 199 | 1945 |
|  query.OrFilterOnRelationshipsAndNested | 30839 | 28939 | 216 | 278 | 1945 |
| 🟥 query.QueryWithNestedIn | 13526 | 12153 | 74 | 80 | 1610 |
|  query.NestedConnectionWhere | 8794 | 8794 | 53 | 85 | 2174 |
|  query.DeeplyNestedConnectionWhere | 8862 | 8844 | 74 | 97 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 6251 | 6251 | 125 | 146 | 1134 |
|  query.DeeplyNestedWithRelationshipSingleFilters | 15438 | 15438 | 91 | 113 | 2174 |
|  query.Fulltext | 96 | 96 | 26 | 29 | 16 |
|  query.FulltextWithNestedQuery | 571 | 571 | 46 | 47 | 84 |
|  sorting.SortMultipleTypes | 2493 | 2493 | 63 | 76 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1498 | 1489 | 181 | 293 | 1040 |
|  sorting.SortOnNestedFields | 13042 | 13042 | 71 | 111 | 2174 |
|  sorting.SortDeeplyNestedFields | 38965 | 38965 | 268 | 201 | 4516 |
|  sorting.SortWithTopLevelCypher | 2119 | 2119 | 188 | 198 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 58 | 76 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 57 | 92 | 35 |
|  unions.NestedUnion | 398553 | 398553 | 300 | 348 | 33033 |
|  unions.NestedUnionWithMissingFields | 372527 | 372527 | 311 | 338 | 33033 |
|  update.NestedUpdate | 2119 | 2119 | 55 | 44 | 1040 |

</details>
        
Old Schema Generation: 33.129s
Schema Generation: 33.356s

File: packages/graphql/src/schema/subscriptions/generate-subscription-where-type.ts [hunk 1]
```diff
@@ -106,25 +107,36 @@ function getRelationshipConnectionWhereTypes({
...
+            fields,
         });
+        acc[fieldName] = relationFieldInputWhereType;
+        return acc;
+    }, {});
+
+    if (isEmptyObject(relationsFieldInputWhereTypeFields)) {
+        return;
+    }
+    // TODO: getORCreate??
```
Comment (angrykoala) [inline packages/graphql/src/schema/subscriptions/generate-subscription-where-type.ts:136]: :eyes:

**Review by angrykoala — APPROVED** [review]
A couple of minor code improvements but good overall :+1:

**Pull request merged**